### PR TITLE
Check eldest KID in ComputeKeyDiffs, fix ugly warning messages when eldest key not pgp key

### PR DESCRIPTION
**Pull request by patrickxb**
r? @maxtaco 

Fixes https://keybase.atlassian.net/browse/CORE-2112 and also fixes https://keybase.atlassian.net/browse/CORE-231

### CODE REVIEW

File: go/libkb/identify_state.go [hunk 1]
```diff
@@ -91,6 +91,17 @@ func (s *IdentifyState) ComputeKeyDiffs(dhook func(keybase1.IdentifyKey)) {
 		dhook(k)
 	}
 
+	// first check the eldest key
+	observedEldest := s.u.GetEldestKID()
+	if s.track != nil {
+		trackedEldest := s.track.GetEldestKID()
+		if observedEldest != trackedEldest {
```
Comment (maxtaco) [inline go/libkb/identify_state.go]: can you use `KID.Eq`?

**Comment by maxtaco** [comment]
I can't seem to repro the original issue right now. Does this patch fix the warning for you?

File: go/libkb/identify_state.go (hunk 1 above)
Comment (patrickxb) [inline go/libkb/identify_state.go]: done

**Comment by patrickxb** [comment]
To repro original warning, track a new user (with nacl eldest key) twice.

Yes, this fixes that.

**Comment by maxtaco** [comment]
Ok, great then!! :+1:

**Pull request merged**